src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -265,60 +251,51 @@ void xma_thread1() {
 void xma_thread2() {
     bool expected = false;
     bool desired = true;
-    int32_t ret = 0;
+    int32_t session_index = 0;
+    int32_t num_sessions = -1;
     while (!g_xma_singleton->xma_exit) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(30));
-        for (auto& itr1: g_xma_singleton->all_sessions) {
+        num_sessions = g_xma_singleton->all_sessions_vec.size();
+        if (num_sessions == 0) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(30));
+            continue;
+        }
+        if (session_index >= num_sessions) {
```

Comment:
This innocent looking line of code is important for it can prevent thread starvation from occurring.  It should have a comment similar to:
`// Using round robin algorithm, go back to the beginning.`

I mentioned this for "someone" in the future my think they could optimize this loop by always setting the session_index to 0 at the beginning of the while loop.  Which would be VERY difficult to debug when things go wrong.